### PR TITLE
austin.intermediate.summer2.rodriguez.stephanie

### PR DESCRIPTION
**Pull request by stephorozcobuilds**
homework/01LessonOne.js done

### CODE REVIEW

File: homework/01LessonOne.js [hunk 1]
```diff
@@ -106,7 +107,7 @@ var notNotTrue;
 // What is the difference between the "and" and the "or" operators? Explain why you would use each of them.
 //
 // Your Answer Goes Here:
-//
+// AND (&&) will test the value on the left and right side for a true value if either side contains false it will return false. OR (||) will test the value on either side is true and if either or both are true it will return true. They are logical operators used to determine the logic between variables or values.
```
Comment (pieroapretto) [inline homework/01LessonOne.js:110]: Great answer!

File: homework/01LessonOne.js [hunk 2]
```diff
@@ -153,47 +154,47 @@ var notNotUndefined;
```
Comment (pieroapretto) [inline homework/01LessonOne.js:154]: Make sure to include an answer to this question:
 Explain truthiness and falsiness in your own words. Provide an example for each.

File: homework/01LessonOne.js [hunk 3]
```diff
@@ -186,7 +186,7 @@ var stringFourEqualsNumberFour = true;
 
 // Problem 29:
 // What is the value of '4' != 4
-var stringFourNotEqualsNumberFour = true;
+var stringFourNotEqualsNumberFour = false;
 
 // Problem 30:
 // What is the value of '4' === 4
```
Comment (pieroapretto) [inline homework/01LessonOne.js:192]: Good corrections here!

**Comment by paulrenenichols** [comment]
@Stephrod Where is your pig latin code?

You can find my code [here](https://gist.github.com/paulrenenichols/099c38009a569d603a098060f6cc3018)

File: apps/02RockPaperScissors.js [hunk 4]
```diff
@@ -8,17 +8,39 @@ prompt.start();
...
+
+   else {
+     if (hand2 === 'paper') {
+       return 'Hand one wins!';
+     }
+     else {
+       return 'Hand two wins';
+     }
+   }
+ }
```
Comment (pieroapretto) [inline apps/02RockPaperScissors.js:40]: Good formatting here! Easy to read.

good job

File: apps/02RockPaperScissors.js [hunk 5]
```diff
@@ -8,17 +8,39 @@ prompt.start();
...
+     if (hand2 === 'rock') {
+       return 'Hand one wins!';
+     }
+     else {
+       return 'Hand two wins!';
+     }
+   }
+
+   else {
+     if (hand2 === 'paper') {
```
Comment (pieroapretto) [inline apps/02RockPaperScissors.js:33]: Sound logic here. 

You could also combine all Hand #1 winning combinations into one 'else if' statement using the **||** operator.

File: homework/03LessonThree.js [hunk 6]
```diff
@@ -20,16 +20,17 @@ var firstName = 'susan';
...
+var valueOfLastName = 'plaid';
 
 // Problem 3:
 // Fix the following function so that it returns the first argument
+var firstArgument = 'bob';
+
 function returnTheFirstArgument(firstArgument) {
-  var firstArgument = 'bob';
   return firstArgument;
 }
```
Comment (ASTPMeetup) [inline homework/03LessonThree.js]: Exactly what we're looking for here. No additional code needed.

Comment (pieroapretto) [inline homework/03LessonThree.js:35]: Exactly what we're looking for here. No additional code needed.

File: homework/03LessonThree.js [hunk 7]
```diff
@@ -41,7 +42,7 @@ function returnTheFirstArgument(firstArgument) {
 // What is scope?
 //
 // Your Answer Goes Here:
-//
+// Scope is a variable that is accessible outside or inside a function. A variable that is declared outside a function definition is a global variable. A variable that is declared inside a function definition is local.
 //
```
Comment (pieroapretto) [inline homework/03LessonThree.js:46]: Good examples here. Scope refers to whether a variable is global, local, etc.

File: homework/03LessonThree.js [hunk 8]
```diff
@@ -89,17 +91,18 @@ var problemSevenAnswer;
 
 // Problem 8:
 // Create an empty array
-var emptyArray;
+var emptyArray = '';
 
 // Problem 9:
 // Create an array with 5 elements in it
-var lengthFiveArray;
+var lengthFiveArray = ['stephanie', 'rodriguez', '2', '3', '4'];
```
Comment (pieroapretto) [inline homework/03LessonThree.js:99]: Nice job adding various data types here.

File: homework/03LessonThree.js [hunk 9]
```diff
@@ -78,8 +79,9 @@ var problemSevenAnswer;
 //
 // What is the ternary operator?  How does it differ from normal conditional statements?
 //
-// Your Answer Goes Here:
 //
+// Your Answer Goes Here:
+// It is an operator that is used as a shortcut for if/else statements.
 //
```
Comment (pieroapretto) [inline homework/03LessonThree.js:85]: Yes! If/else statement between two arguments or values.

File: homework/03LessonThree.js [hunk 10]
```diff
@@ -110,10 +113,15 @@ var threeByThreeArray = [
...
+var threeByThreeArray = [
+  [1, 2, 3],
+  [4, 10, 6],
+  [7, 8, 9]
+];
 
 // Problem 12:
 // create a 2 by 2 (2 rows, 2 columns) nested array
-var twoByTwoArray;
+var twoByTwoArray = [['row1','row2'],['column1','column2']];
```
Comment (pieroapretto) [inline homework/03LessonThree.js:125]: Great job passing all your mocha tests!

wow2

File: apps/03TicTacToe.js [hunk 11]
```diff
@@ -23,23 +23,57 @@ function printBoard() {
...
+    return (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+    (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+    (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn)
+};
 
 function diagonalWin() {
     // Your code here
-}
+    return (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+    (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)
```
Comment (pieroapretto) [inline apps/03TicTacToe.js:41]: Great job not using else/if statements and simply returning the Boolean value!

File: apps/03TicTacToe.js [hunk 12]
```diff
@@ -23,23 +23,57 @@ function printBoard() {
...
-    // Your code here
-}
+    if (board[row][column] === 'X' || board[row][column] === 'O' ) {
+      console.log('Invalid selection player ' + playerTurn + ' must select an empty space!');
+    }
+    else {
+        board[row][column] = playerTurn;
+        checkForWin();
+        playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+    }
```
Comment (pieroapretto) [inline apps/03TicTacToe.js:74]: Really clean logic here! You must have a great TA haha

File: homework/04LessonFour.js [hunk 13]
```diff
@@ -14,51 +14,69 @@ var numberArray = [1, 2, 3, 4, 5];
...
+};
 
 // Problem 9:
 // DON'T EDIT problemNineObject!!!!
 var problemNineObject = {
-  'fav food': 'pizza',
-  city:       'houston'
+  'fav food': 'tacos',
+  city:       'austin'
 };
```
Comment (pieroapretto) [inline homework/04LessonFour.js:80]: Looks like you directly edited the contents of problems 9 and 10.

Use your superior knowledge of associative arrays to change the values for each nested object provided.

Example:  _nestedObjects.users['bob roberts'].starring = 'Tim Robbins';_

File: homework/04LessonFour.js [hunk 14]
```diff
@@ -87,7 +105,7 @@ var nestedObjects = {
 // What is an associative array? What is the difference between an array and an associative array?
 //
 // Your Answer Goes Here:
-//
+// An associative array is a collection of of data called a key, the key has an index that is assigned specific values. Arrays have an index that start at specifically at 0. 
 //
```
Comment (pieroapretto) [inline homework/04LessonFour.js:113]: You have the right idea here.

File: homework/04LessonFour.js [hunk 15]
```diff
@@ -87,7 +109,7 @@ var nestedObjects = {
 // What is an associative array? What is the difference between an array and an associative array?
 //
 // Your Answer Goes Here:
-//
+// An associative array is a collection of of data called a key, the key has an index that is assigned specific values. Arrays have an index that start at specifically at 0.
```
Comment (pieroapretto) [inline homework/04LessonFour.js:112]: Your mind is in the right place here. An associative array is a collection of named keys that each contain data/value.

File: homework/05LessonFive.js [hunk 16]
```diff
@@ -63,15 +63,15 @@ for (forLoopCounter = 0; forLoopCounter > -10; forLoopCounter--) {
...
+var forLoopArrayResult = [0, -1, -2, -3, -4, -5, -6, -7, -8, -9];
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What are loops? Why do we use them?
-//
+//Loops are set of instructions used to repeat the same block of code till a specified condition returns false or true depending on how you need it.
```
Comment (pieroapretto) [inline homework/05LessonFive.js:74]: whoa

File: apps/05Mastermind.js [hunk 17]
```diff
@@ -9,37 +9,78 @@ var board = [];
...
+
+    for (var i = 0; i < solutionArray.length; i++) {
+      if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++;
+        solutionArray[i] = null;
+      }
+    }
+    var correctLetters = 0;
+    for (var i = 0; i < solutionArray.length; i++) {
+      if (guessArray.indexOf(solutionArray[i])) {
```
Comment (pieroapretto) [inline apps/05Mastermind.js:44]: Your two 'if' statements in this for loop seem repetitive..

File: homework/06LessonSix.js [hunk 18]
```diff
@@ -76,27 +83,32 @@ function ConferenceRoom() {
...
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What is a method?
-//
+// A Method is a function associated with an Object (a Constructor Function or Class)
+// and a Method function has access to the data present in its Construction function.
```
Comment (pieroapretto) [inline homework/06LessonSix.js:111]: Great answer!

File: apps/05Mastermind.js [hunk 19]
```diff
@@ -9,37 +9,78 @@ var board = [];
...
-}
+    if (board.length < 10) {
+      if (guess === solution) {
+        return 'You guessed it!';
+      }
+      else {
+        board += 1;
+        return generateHint(solution, guess);
+      }
+    }
```
Comment (pieroapretto) [inline apps/05Mastermind.js:67]: What does mastermind() return if board.length is greater or equal to 10??

File: homework/06LessonSix.js [hunk 20]
```diff
@@ -76,27 +83,32 @@ function ConferenceRoom() {
 // Add a method called sayHi() that returns "Hi, I'm " + this.name + '!';
 function Person(name) {
   this.name = name;
+  this.sayHi = function () {
+    return "Hi, I'm " + this.name + '!';
```
Comment (pieroapretto) [inline homework/06LessonSix.js:87]: Nice!

File: apps/06SpaceTravelToMars.js [hunk 21]
```diff
@@ -10,6 +10,38 @@ var jobTypes = {
...
+function Ship(name, type, ability) {
+  this.name = name;
+  this.type = type;
+  this.ability = ability;
+  this.crew = [];
+
+  this.missionStatement = function() {
+    for (var i = 0; i < this.crew.length; i++) {
+      var currentCrewmember = this.crew[i].job;
+      if (jobTypes[currentCrewmember]=== this.type) {
```
Comment (pieroapretto) [inline apps/06SpaceTravelToMars.js:34]: Very nice!

Comment (pieroapretto) [inline apps/06SpaceTravelToMars.js:34]: Very nice!

File: homework/07LessonSeven/script.js [hunk 22]
```diff
@@ -69,20 +69,21 @@ $(document).on('ready', function() {
...
   // select the element with id="data-element"
   // and set window.dataIdValue to the value of the data-id attribute of the element
   // you can do this by calling .data() on the selected element
-  window.dataIdValue;
+  window.dataIdValue = $('#data-element').data('id');
 
   // Problem 12:
   // select the element with id="data-element"
   // and set its text (.text()) to the value of the data-content attribute
+  $('#data-element').text($('#data-element').data('content'));
```
Comment (pieroapretto) [inline homework/07LessonSeven/script.js]: Good job chaining these jQuery methods. For readability, I recommend breaking up these kind of commands into parts in the future.

File: apps/07jquery-tic-tac-toe/script.js [hunk 23]
```diff
@@ -2,4 +2,63 @@
...
+      checkForWin();
+      playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+
+    });
+
+    $("#clear").click(function() {
+    $('[data-cell]').each(function(){
+        $(this).empty();
+    });
+  $("#announce-winner").empty();
```
Comment (pieroapretto) [inline apps/07jquery-tic-tac-toe/script.js:25]: Good job clearing out this div after every click.

File: apps/07jquery-tic-tac-toe/script.js [hunk 24]
```diff
@@ -2,4 +2,63 @@
...
+
+    function diagonalWin() {
+        // Your code here
+        return $('[data-cell="0"]').text() === playerTurn &&
+        $('[data-cell="4"]').text() === playerTurn &&
+        $('[data-cell="8"]').text() === playerTurn ||
+        $('[data-cell="2"]').text() === playerTurn &&
+        $('[data-cell="4"]').text() === playerTurn &&
+        $('[data-cell="6"]').text() === playerTurn
+    };
```
Comment (pieroapretto) [inline apps/07jquery-tic-tac-toe/script.js:63]: Good formatting in this app. Good job!

File: homework/08LessonEight/script.js [hunk 25]
```diff
@@ -17,30 +17,30 @@ $(document).on('ready', function() {
...
-  $finalList;
+  $finalList.prepend(window.firstListChildren);
 
   // Problem 6:
   // use $finalList.append() to put window.secondListChildren
   // at the end of #final-list
-  $finalList;
+  $finalList.append(window.secondListChildren);
 
 });
```
Comment (pieroapretto) [inline homework/08LessonEight/script.js:46]: goodjob

**Pull request closed without merging**